docs/core/deploying/prepare-libraries-for-trimming.md

```diff
@@ -177,6 +177,43 @@ static void InitializeTypeField()
 
 In this case the trim analysis will simply keep public methods of `System.Tuple`, and will not produce further warnings.
 
+### DynamicDependency
+
+This attribute can be used to indicate that a member has a dynamic dependency on other members. This results in the referenced members being kept whenever the member with the attribute is kept, but doesn't silence warnings on its own.
+This can be used together with `UnconditionalSuppressMessageAttribute` to fix some analysis warnings.
```

Comment:
I would add a warning like:
Usage of `DynamicDependencyAttribute` should be limited to cases where no other approach is usable. Trimming tools can't validate that the dependency specified by the attribute will be enough to make the application work, so it's very likely trimming will produce warnings even if the usage of the attribute fully covers the use case. If such warnings are suppressed (via `UnconditionalSuppressMessageAttribute`) trimmer will also not validate that potential future changes in the code are correctly reflected in the usage of `DynamicDependencyAttribute`.

Reply:
In short - we don't want people to use this - unless they absolutely have to.

Reply:
I moved this down to the "advanced" section and added some warnings - PTAL.